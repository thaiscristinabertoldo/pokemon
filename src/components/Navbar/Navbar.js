import React, { memo } from 'react'

import { Link } from 'react-router-dom'

import NavbarStyled from './navbar.css'

import logo from '../../assets/logo.png'

export const Navbar = memo(() => {
  return (
    <NavbarStyled data-testid="navbar">
      <Link data-testid="home-link" to="/pokemon">
        <img className="img" src={logo} alt="PokÃ©mon" height="50px" />
      </Link>
    </NavbarStyled>
  )
})
