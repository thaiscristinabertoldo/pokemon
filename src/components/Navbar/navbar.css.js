import styled from 'styled-components'

const NavbarStyled = styled.div`
  overflow: hidden;
  position: fixed;
  width: 100%;
  top: 0;
  padding: 10px;
  background-color: #000000;
  z-index: 1;
  .img {
    cursor: pointer;
  }
`

export default NavbarStyled
