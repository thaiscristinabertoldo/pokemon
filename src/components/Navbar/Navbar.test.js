import React from 'react'

import { Router } from 'react-router-dom'

import { render, cleanup, fireEvent } from '@testing-library/react'

import { createMemoryHistory } from 'history'

import { Navbar } from './Navbar'

afterEach(cleanup)

const renderWithRouter = (component) => {
  const history = createMemoryHistory()
  return {
    ...render(<Router history={history}>{component}</Router>),
  }
}

describe('Navbar', () => {
  it('should take a snapshot', () => {
    const { asFragment } = renderWithRouter(<Navbar />)

    expect(asFragment(<Navbar />)).toMatchSnapshot()
  })

  it('should render the navbar link', () => {
    const { getByTestId } = renderWithRouter(<Navbar />)
    const navbar = getByTestId('navbar')
    const link = getByTestId('home-link')

    expect(navbar).toContainElement(link)
  })

  it('should navigate link', () => {
    const { getByTestId } = renderWithRouter(<Navbar />)

    fireEvent.click(getByTestId('home-link'))
  })
})
