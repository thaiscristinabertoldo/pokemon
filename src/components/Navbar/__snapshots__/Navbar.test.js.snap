// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm hYyltE"
    data-testid="navbar"
  >
    <a
      data-testid="home-link"
      href="/pokemon"
    >
      <img
        alt="PokÃ©mon"
        class="img"
        height="50px"
        src="logo.png"
      />
    </a>
  </div>
</DocumentFragment>
`;
